### PR TITLE
[17.0][MIG] github_connector_odoo: Migration to 17.0

### PR DESCRIPTION
**Pull request by victoralmau**
Migration to 17.0

@Tecnativa TT51639

### CODE REVIEW

**Comment by pedrobaeza** [comment]
You can rebase.

**Comment by victoralmau** [comment]
The error in the tests is a reflection of a curious thing, the `adapt_version()` method has changed from [v16](https://github.com/odoo/odoo/blob/16.0/odoo/modules/module.py#L527) to [v17](https://github.com/odoo/odoo/blob/17.0/odoo/modules/module.py#L447) causing it to now fail. What solution do you propose?

**Comment by victoralmau** [comment]
It's ready to review now.

File: github_connector_odoo/monkeypatching.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+import odoo.release as release
+from odoo.modules import module
+
+adapt_version_original = module.adapt_version
+
+
+def adapt_version(version):
+    """Overwrite the method to avoid the error if use values that do not contain serie
+    (17.0), for example 13.0.4.0.0.0
```
Comment (pedrobaeza) [inline github_connector_odoo/monkeypatching.py]: I think the example should `4.0`, not a complete version containing even more 0s.

File: github_connector_odoo/views/view_github_repository_branch.xml [hunk 2]
```diff
@@ -51,7 +51,7 @@ License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
                     name="module_version_analysis_rule_info_ids"
                     widget="one2many_list"
                     colspan="4"
-                    attrs="{'invisible': [('module_version_analysis_rule_info_ids', '=', [])]}"
+                    invisible="module_version_analysis_rule_info_ids==[]"
```
Comment (pedrobaeza) [inline github_connector_odoo/views/view_github_repository_branch.xml]: Is this working this way?

**Review by pedrobaeza — APPROVED** [review]

**Review by pilarvargas-tecnativa — APPROVED** [review]

**Pull request merged**